src/components/common/note-loading-boundary/create-non-existing-note-hint.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { Trans, useTranslation } from 'react-i18next'
+import React, { useCallback } from 'react'
+import { Alert, Button } from 'react-bootstrap'
+import { useSingleStringUrlParameter } from '../../../hooks/common/use-single-string-url-parameter'
+import { createNoteWithPrimaryAlias } from '../../../api/notes'
+import { useAsyncFn } from 'react-use'
+import { ShowIf } from '../show-if/show-if'
+import { Redirect } from '../redirect'
+import { ForkAwesomeIcon } from '../fork-awesome/fork-awesome-icon'
+import { testId } from '../../../utils/test-id'
+
+export const CreateNonExistingNoteHint: React.FC = () => {
+  useTranslation()
+  const noteIdFromUrl = useSingleStringUrlParameter('noteId', undefined)
+
+  const [returnState, createNote] = useAsyncFn(async () => {
+    if (noteIdFromUrl === undefined) {
+      throw new Error('Note id not set')
+    }
+    return await createNoteWithPrimaryAlias('', noteIdFromUrl)
+  }, [noteIdFromUrl])
+
+  const onClickHandler = useCallback(() => {
+    void createNote()
+  }, [createNote])
+
+  if (noteIdFromUrl === undefined) {
+    return null
+  } else if (returnState.value) {
+    return <Redirect to={`/n/${returnState.value.metadata.primaryAddress}`} />
+  } else if (returnState.loading) {
+    return (
+      <div className={'d-flex flex-column mt-5'} {...testId('loadingMessage')}>
+        <b>
+          <ForkAwesomeIcon icon={'spinner'} className={'fa-spin mr-2'} />
+          <Trans i18nKey={'noteLoadingBoundary.createNote.creating'} />
+        </b>
+      </div>
+    )
+  } else if (returnState.error !== undefined) {
+    return (
+      <Alert variant={'danger'} {...testId('failedMessage')}>
+        <ForkAwesomeIcon icon={'exclamation-triangle'} className={'mr-1'} />
+        <Trans i18nKey={'noteLoadingBoundary.createNote.error'} />
+      </Alert>
+    )
+  } else {
+    return (
+      <div className={'d-flex flex-column mt-5'}>
+        <b>
+          <Trans i18nKey={'noteLoadingBoundary.createNote.question'} values={{ aliasName: noteIdFromUrl }} />
```

Comment:
Is the content of `values` cached properly for re-renders, as this is an object and not a plain data type?

Reply:
In this case it should be okay because this part of the component doesn't get rerendered that often.

Reply:
Okay. When we do performance measurements at some point, we could take a look again.